### DISCUSSION TITLE
LXQt-labwc-session repository

### DISCUSSION THREAD

**Discussion by stefonarch** [labwc/labwc#967] [Show and tell]
I've set up[ this repo](https://github.com/stefonarch/lxqt-labwc-session) as labwc's window rules support now much of LXQt, atm it's still private as it needs more testing and hopefully some feedback from invited people. After that it could be in AUR.

It adds everything to have a basic wayland session which can be started from sddm or other DMs and should work in parallel with existing x11 LXQt sessions.

**Comment by johanmalm** [comment]
I've not had a chance to play with it yet. Background looks cool though :smile:

**Reply by stefonarch** [reply]
BTW there are no bigger graphics anywhere? It's a bit blurry as it comes from a 200px png file...

**Comment by stefonarch** [comment]
There is one thing I'd like to get some thoughts on: my setup atm uses `~/.config/lxqt-wayland/labwc/` for the settings, with the `-C` option. This makes sure there is no interference with existing configs but has the downside of doubling everything maybe already configured.

Another options are:

*  use only `~/.config/labwc/lxqt-rc.xml`.
* write the window rules and at least the runner shortcut to the default file.
* others?

There will be users who have already knowledge of labwc and those who will run it the first time, for the latter "there is no second opportunity to make a first impression". For the first type of users their "autostart" file maybe start things which are also started by LXQt in `~/.config/autostart`.

As the position of the config files should  never be changed afterwards so this needs attention IMO.

**Reply by johanmalm** [reply]
This might need some thought.

IIRC XDG spec says that we use config files in XDG_CONFIG_DIRS first and the let the user override with anything in XDG_CONFIG_HOME. 

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#:~:text=%24XDG_CONFIG_HOME%20defines%20the%20base%20directory,state%20files%20should%20be%20stored.

I think this type of approach is what we should explore first. I can remember how compliant we are that approach off the top of my head.

Also I can't remember if we are allowed to read from DIRS first and the overtide with anything set in HOME. Or whether it’s an exclusive logic.

**Comment by stefonarch** [comment]
Looks like there is was no link to the repo, added it in the first posting too:
https://github.com/stefonarch/lxqt-labwc-session

Arch users can use the [AUR package](https://aur.archlinux.org/packages/lxqt-labwc-session-git).

Screenshots:

![labwc-dark](https://github.com/labwc/labwc/assets/10681413/57028487-c196-4e35-80b8-d99665586348)

![labwc-light](https://github.com/labwc/labwc/assets/10681413/f9f45e4e-2ba3-4ec3-9dba-3006311d64b5)

**Reply by Consolatis** [reply]
Nice screenshots. Maybe you could add them to your repo README as well?

**Reply by stefonarch** [reply]
I made them for the repo actually  yesterday, testing again a new user,  then I stumbled upon this discussion.
Have also to fix the setting for the  icon loading on top left.

And by testing the new user I noticed [again](https://github.com/labwc/labwc/issues/945) 3 cursors:
* lxqt-panel: white, but bigger
* labwc white
* waybar: small, black 

The last one  depends  how  GTK styling is handled, by default there is no override of it in LXQt.

**Reply by Consolatis** [reply]
We are using the `XCURSOR_THEME` and `XCURSOR_SIZE` env vars. Not sure how other frameworks / applications decide which cursor theme (and size) to load.

**Reply by stefonarch** [reply]
I know, but I remember going mad about this in June. I see in the clean user the LXQt setting:
```
 [Mouse]
 cursor_size=30
 cursor_theme=Breeze_Snow
```
So basically those need to be in sync somehow, otherwise cursor size and theme can change.

**Reply by Consolatis** [reply]
Maybe the startup script could parse the LXQt setting and then modify the environment file appropriately? Or alternatively, export them as env vars before starting labwc (and removing them from the environment file completely). That should at least make labwc honor the LXQt settings.

**Reply by stefonarch** [reply]
Probably the best, like it does now for the keyboard layout.

**Reply by stefonarch** [reply]
Done in https://github.com/stefonarch/lxqt-labwc-session/commit/5800f0a7ae945a35fb21a0d627e36f974f8803a2

Cursor size and theme are now set by `lxqt-config-appearance` perfectly.

**Comment by stefonarch** [comment]
There is some real  nice progress now in Qt6/KF6 with `layer-shell-qt` v6.

[Here](https://share.mailbox.org/ajax/share/0080f490003dac420d9dbc703dac45b28629d4b1f94e50c2/1/8/NDA/NDAvMTA) is a demo with kwin labwc and sway, atm taskbar (and more) works only on kwin and  layer-shell code has to be implemented in other components.

**Reply by johanmalm** [reply]
Nice! Thanks for sharing.

**Reply by johanmalm** [reply]
I've had a quick play with `layer-shell-qt` at `>=6.5`. Seems to work well.

**Reply by ffs-akl** [reply]
@stefonarch  I saw you running labwc with xwayland in vmware, didn't labwc crash when run a xwayland program?
Because labwc crashes in vm, I had to disable xwayland when compiling labwc

**Reply by stefonarch** [reply]
Hm, didn't test this, could be.
The only xwayland app I use regularly is element-desktop but on real hardware.

**Comment by stefonarch** [comment]
Updated for LXQt 2.0 now. Mostly removed window rules :)
https://github.com/stefonarch/lxqt-labwc-session

**Reply by johanmalm** [reply]
Nice.

I've installed v2 of `pcmanfm-qt` + `lxqt-panel` too and they work well. Nice to get rid of those window-rules.

**Reply by ffs-akl** [reply]
did the lastest lxqt-panel (2.0.1-1) in arch extra have no problem with taskbar ?

**Reply by stefonarch** [reply]
It has no  taskbar on wayland yet. 
If you want it (in development, usable but has some minor bugs) install this PKBUILD: https://github.com/stefonarch/LXQt-Wayland-files/tree/main/AUR/lxqt-panel

**Reply by ffs-akl** [reply]
@stefonarch  I've tried both aur and kwin_wayland only  today, but both of them have not been successfully compiled.  Report an error at 70-80% of compilation

**Reply by stefonarch** [reply]
It works for me and others, what error did you get?

**Reply by ffs-akl** [reply]
it would be possible to send your binary package to me.  I see that the kwin_wayland only git branch is a non-compilable statu.
![swappy-20240513-085917](https://github.com/labwc/labwc/assets/166389780/a7ce65f5-1fe9-41dc-ba1f-a3ce3fc7a9b8)

**Reply by ffs-akl** [reply]
@stefonarch 
There is  two error ( VMWARE+Arch+Palsma 6):
/home/knm/aur/src/lxqt-panel/plugin-taskbar/lxqttaskbarplugin.h:62:1: error: Undefined interface

make[2]: *** [plugin-taskbar/CMakeFiles/taskbar_autogen.dir/build.make:75：plugin-taskbar/taskbar_autogen/timestamp] error 1
make[1]: *** [CMakeFiles/Makefile2:2063：plugin-taskbar/CMakeFiles/taskbar_autogen.dir/all] error 2
make: *** [Makefile:136：all] error 2

**Reply by tokyo4j** [reply]
I remember I got the same error as @knm100's.

**Reply by stefonarch** [reply]
Strange,  no idea, both PRs build here, the CI error is not relevant as it's broken atm.
 This error was reported and fixed, but it's not at build time: https://github.com/lxqt/lxqt-panel/issues/2052

I remember now that some files aren't included anymore  which are included in the PR taskbar-refactor, could be this, but as it's a PR depending on 2 others (one beeing reviewed now and merged soon) there could be issues hard to track.

I can't attach the Arch package, it's [here](https://share.mailbox.org/ajax/share/004671380491344a01f5e6f49134423ab2e0979d9747eda1/1/8/NDI/NDIvMTI).

EDIT: this is a package for my use which doesn't check dependencies and conflicts. I'll replace it.

**Reply by ffs-akl** [reply]
Thank you !

**Reply by stefonarch** [reply]
I've forgotten about all the patches ;)
This package contains some fixing commit, I pushed it here now: https://github.com/stefonarch/lxqt-panel/tree/wlroots_taskbar if somebody likes to try if that compiles well.

**Reply by ffs-akl** [reply]
There is one warning and two errors
/home/knm/aur/src/lxqt-panel/plugin-desktopswitch/desktopswitch.cpp: In member function ‘virtual void DesktopSwitch::settingsChanged()’:
/home/knm/aur/src/lxqt-panel/plugin-desktopswitch/desktopswitch.cpp:254:13: warning：unused variable ‘columns’ [-Wunused-variable]
  254 |         int columns = static_cast<int>(ceil(static_cast<float>(m_desktopCount) / mRows));
      |             ^~~~~~~

/home/knm/aur/src/lxqt-panel/plugin-taskbar/lxqttaskbarplugin.h:62:1: error: Undefined interface

make[2]: *** [plugin-taskbar/CMakeFiles/taskbar_autogen.dir/build.make:75：plugin-taskbar/taskbar_autogen/timestamp] error 1
make[1]: *** [CMakeFiles/Makefile2:2063：plugin-taskbar/CMakeFiles/taskbar_autogen.dir/all] error 2
make: *** [Makefile:136：all] error 2

**Reply by stefonarch** [reply]
Basically the same error, no clue...

**Reply by ffs-akl** [reply]
would you send the binary package in tar.zst format to me?   I downloaded the binary package you uploaded but it is in tar format. Even though I repackaged it into tar.zst format, I still can't install it via “pacman -U”.

**Reply by stefonarch** [reply]
Hm, would have to rebuild it changing my settings I guess. In `/etc/makepkg.conf`
```
#########################################################################
# EXTENSION DEFAULTS
#########################################################################
#
PKGEXT='.pkg.tar'
SRCEXT='.src.tar.gz'
```
It's ages I use `pkg.tar`, no idea why you can't install it with `-U`.

**Comment by romildo** [comment]
I am also getting the error on NixOS:
```console
Output
------
/build/source/plugin-taskbar/lxqttaskbarplugin.h:62:1: error: Undefined interface

make[2]: *** [plugin-taskbar/CMakeFiles/taskbar_autogen.dir/build.make:75: plugin-taskbar/taskbar_autogen/timestamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:2009: plugin-taskbar/CMakeFiles/taskbar_autogen.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by stefonarch** [comment]
Ok, it's again this strange thing about 4 files needed and not found by the compiler. You'll need to copy manually
```
ilxqtpanel.h
ilxqtpanelplugin.h  
lxqtpanelglobals.h  
pluginsettings.h
``` 
from `lxqt-panel/panel` to `/usr/include/lxqt/` before compiling. Probably having installed already `lxqt-panel` would work too.
